### ISSUE TITLE
Find a way to automatically build and push x86_64 and aarch64 images

### ISSUE THREAD

**Issue by ericcurtin** [containers/ramalama#27]
quay.io can only automatically build x86_64

aarch64 is important, this is tested to work on Apple Silicon/macOS with podman machine and libkrun.

we just need to figure out a way to push aarch64 images.

**Comment by ericcurtin** [comment]
@rhatdan this is essential for the podman machine use case on macOS

https://github.com/ericcurtin/ramalama/issues/14

Migrated from above, @cooktheryan has an idea how to do it...

I don't know how to automate pushing multi-arch container builds to github

**Comment by rhatdan** [comment]
@cevich has set something up on Podman to do this automatically I believe using a systemd-timer or cron job.

Chris any pointers on how to do this?

**Comment by cevich** [comment]
Yes, and for Buildah and Skopeo and an AIO image as well.  The implementation is over in https://github.com/containers/image_build but it relies on Cirrus-CI and private VM resources.  At it's core is [a bash script](https://github.com/containers/automation/tree/main/build-push) that may/not be fit-for-purpose here.

If your builds are basic, I would suggest your they could be done using [the Buildah github action](https://github.com/redhat-actions/buildah-build).  However, be warned that the containers-org is operating near the free-tier limit of 20-runners, so jobs may be stalled or delayed.  If you have a way to provide your own runners (for this repo) that would be great.  Otherwise it should probably be fine with (for example) once-per-day builds but I can't guarantee anything long-term :disappointed:

**Comment by ericcurtin** [comment]
@cooktheryan 

I want to open a PR that does something like this:

```
RUN    if [ "$(uname -m)" = "aarch64" ]; then \
      dnf copr enable -y slp/mesa-krunkit && \
      dnf install -y mesa-libOpenCL-23.3.5-102.aarch64 \
        mesa-vulkan-drivers-23.3.5-102.aarch64 && \
      dnf versionlock mesa-libOpenCL-23.3.5-102.aarch64 \
        mesa-vulkan-drivers-23.3.5-102.aarch64; \
    fi && \
```

one thing that could be super helpful is to build both for aarch64 and x86_64 per PR (but not push). At the moment we just do x86_64 it would be better to do both to avoid breakages in the nightly build.

Thanks for pushing on the nightly builds btw.

**Comment by rhatdan** [comment]
Care to open a PR to do this on RamaLama?

**Comment by debarshiray** [comment]
It was @travier who did all the heavy lifting and I still have one Podman pull request that I should wrap up before I can get to this one.

In short, you can start by copy pasting one of the YAML files from https://github.com/containers/toolbox/tree/main/.github/workflows and replacing the specifics, like the paths to the image sources, etc..  `QUAY_ROBOT_TOKEN` is added to the GitHub project's settings under *secrets and variables*.  I forgot if there was something that needed to be done on Quay, but I am sure @travier will remember.

In short, it's simple, but one needs to devote some time to iron out the annoying details, and I am struggling to find the time to own that work.  :/

**Comment by ericcurtin** [comment]
@smooge @gnaponie @lmilbaum are on this one

**Comment by rhatdan** [comment]
This is still being worked on.

**Comment by lmilbaum** [comment]
I would be happy to help

**Comment by rhatdan** [comment]
@mikebonnet Please update this issue.

**Comment by mikebonnet** [comment]
We are building aarch64 images in Konflux and will be pushing them to the quay.io repos very soon (in the next week or so). I'll post an update here when they're available for testing.

**Comment by rhatdan** [comment]
We are now building images in Konflux, closing.
